holoviews/plotting/bokeh/callbacks.py

```diff
@@ -152,39 +164,84 @@ def __init__(self, plot, streams, source, **params):
         self.streams = streams
         self.comm = self._comm_type(plot, on_msg=self.on_msg)
         self.source = source
+        self.handle_ids = defaultdict(list)
 
 
     def initialize(self):
         plots = [self.plot]
         if self.plot.subplots:
             plots += list(self.plot.subplots.values())
 
+        handles = {}
+        for plot in plots:
+            for k, v in plot.handles.items():
+                if k not in handles:
+                    handles[k] = v
+        self.handle_ids.update(self._get_handle_ids(handles))
+
         found = []
         for plot in plots:
             for handle in self.handles:
                 if handle not in plot.handles or handle in found:
                     continue
-                self.set_customjs(plot.handles[handle])
+                self.set_customjs(plot.handles[handle], handles)
```

Comment:
If I understand right, ``found`` is a list of handles that are in some sense 'unique'. So maybe it could be something like:

```
unique_handles = self.unique_handles(plots)
for unique_handle in unique_handles:
   self.set_customjs(plot.handles[handle], handles)
```

Now you might want to call ``unique_handles`` something like ``filtered_handles`` instead and this variable would then replace ``found``.